lib/runner.js

```diff
@@ -834,7 +857,7 @@ Runner.prototype.uncaught = function(err) {
     runnable = new Runnable('Uncaught error outside test suite');
     runnable.parent = this.suite;
 
-    if (this.started) {
+    if (this.state === constants.STATE_RUNNING) {
       this.fail(runnable, err);
     } else {
       // Can't recover from this failure
```

Comment:
This part is probably wrong as well. Why don't we just abort here, or throw? When the runner has stopped, we throw anyway. When it hasn't even started, then there is nothing to recover or fail or to report, so we should just throw.

Reply:
I'm not sure if it's wrong or not.  If we know which `Runnable` is throwing the uncaught error, then I think it makes sense to report it as a failure instead of just bailing.  But I could be convinced otherwise.  Either way, I think we can address it in another PR?

Reply:
Agree, separate PR.